basicswap/bin/prepare.py

```diff
@@ -2639,19 +2659,96 @@ def main():
         logger.info(f"Done. Coin {add_coin} successfully added.")
         return 0
 
-    logger.info("With coins: %s", ", ".join(with_coins))
+    logger.info(
+        "With coins: "
+        + (", ".join(with_coins))
+        + ("" if with_coins_changed else " (default)")
+    )
     if os.path.exists(config_path):
-        if not prepare_bin_only:
-            exitWithError("{} exists".format(config_path))
-        else:
+        if prepare_bin_only:
             with open(config_path) as fs:
                 settings = json.load(fs)
 
                 # Add temporary default config for any coins that have not been added
                 for c in with_coins:
                     if c not in settings["chainclients"]:
                         settings["chainclients"][c] = chainclients[c]
+        elif upgrade_cores:
+            with open(config_path) as fs:
+                settings = json.load(fs)
+
+            with_coins_start = with_coins
+            if not with_coins_changed:
+                for coin_name, coin_settings in settings["chainclients"].items():
+                    with_coins_start.add(coin_name)
+
+            with_coins = set()
+            for coin_name in with_coins_start:
+                if coin_name not in chainclients:
+                    logger.warning(f"Skipping unknown coin: {coin_name}.")
+                    continue
+                current_coin_settings = chainclients[coin_name]
+                if coin_name not in settings["chainclients"]:
+                    exitWithError(f"{coin_name} not found in basicswap.json")
+                coin_settings = settings["chainclients"][coin_name]
+
+                current_version = current_coin_settings["core_version_no"]
+                have_version = coin_settings.get("core_version_no", "")
+
+                current_version_group = current_coin_settings.get(
+                    "core_version_group", ""
+                )
+                have_version_group = coin_settings.get("core_version_group", "")
+
+                logger.info(
+                    f"{coin_name}: have {have_version}, current {current_version}."
+                )
+                if not (
+                    coin_settings.get("manage_daemon", False)
+                    or coin_settings.get("manage_wallet_daemon", False)
+                ):
+                    logger.info("  Unmanaged.")
+                elif have_version != current_version:
```

Comment:
```suggestion
                if have_version != current_version:
```

i think it might be a good idea to update disabled coins as well.

avoids inadvertently missing important updates  -> reenabling and breaking blockchain by running old version on new fork